mlx/random.cpp

```diff
@@ -172,6 +173,48 @@ array normal(
   return samples;
 }
 
+array multivariate_normal(
+    const array& mean,
+    const array& cov,
+    const std::vector<int>& shape,
+    Dtype dtype,
+    const std::optional<array>& key,
+    StreamOrDevice s) {
+  auto stream = to_stream(s);
```

Comment:
Probably I would just require everything to be fp32 here for now (throw otherwise). It really only works for fp32 and I think it's better to be explicit.

Reply:
Do you mean the `dtype` passed to the function? In that case you are right, since all the operations are working just for `float32`. I would keep the possibility of having `mean` and `variance` of other types though.